deploy/chart/templates/tests/test-connection.yaml

```
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ include "cluster-api-state-metrics.fullname" . }}-test-connection"
+  labels:
+    {{- include "cluster-api-state-metrics.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args: ['{{ include "cluster-api-state-metrics.fullname" . }}:{{ .Values.service.port }}']
```

```suggestion
      command:
        - "wget"
      args:
        - "{{ include "cluster-api-state-metrics.fullname" . }}:{{ .Values.service.port }}"
```

I think it's better to use yaml instead of json here